File: DataStructurePython/queue.py
class Queue:
    def __init__(self):
        self.array = []


    def is_empty(self):
        return len(self.array) == 0


    def enqueue(self, item):
        self.array.append(item)


    def dequeue(self):
        if self.is_empty():
            return
        val = self.array[0]
        del self.array[0]
        return val


    def peek_front(self):
        if self.is_empty():
            return
        return self.array[0]

File: DataStructureSwift/Sources/DataStructureSwift/Queue.swift
//
//  Queue.swift
//  
//
//  Created by Yubo Qin on 11/29/22.
//

import Foundation

class Queue<T> {

    private var array: [T]

    init() {
        self.array = []
    }

    var isEmpty: Bool {
        array.isEmpty
    }

    func enqueue(_ item: T) {
        array.append(item)
    }

    @discardableResult
    func dequeue() -> T? {
        if isEmpty {
            return nil
        }
        let first = array.first
        array.remove(at: 0)
        return first
    }

    func peekFront() -> T? {
        return array.first
    }

}

extension Queue: CustomStringConvertible {

    var description: String {
        return "Queue: \(array)"
    }

}
